lib/rules/no-extra-boolean-cast.js

```diff
@@ -35,6 +35,11 @@ module.exports = {
                 enforceForLogicalOperands: {
                     type: "boolean",
                     default: false
+                },
+
+                enforceForInnerExpressions: {
```

Comment:
We could use `oneOf` to prevent specifying both options in the same config and thus avoid confusion (e.g., how is the rule expected to work when `enforceForInnerExpressions` is set to `true` and `enforceForLogicalOperands` explicitly set to `false`).

```js
schema: [{
    anyOf: [
        {
            type: "object",
            properties: {
                enforceForInnerExpressions: {
                    type: "boolean"
                }
            },
            additionalProperties: false
        },

        // depecated
        {
            type: "object",
            properties: {
                enforceForLogicalOperands: {
                    type: "boolean"
                }
            },
            additionalProperties: false
        }
    ]
}],
```

Reply:
Good point! Changed!